core/markets/storage/client/impl/storage_market_client_impl.cpp

```diff
@@ -481,16 +573,21 @@ namespace fc::markets::storage::client {
       ClientEvent event,
       StorageDealStatus from,
       StorageDealStatus to) {
-    // todo validate deal published
-    OUTCOME_EXCEPT(fsm_->send(deal, ClientEvent::ClientEventDealPublished));
+    auto verified = verifyDealPublished(deal);
+    FSM_HALT_ON_ERROR(verified, "Cannot get publish message", deal);
+    if (!verified.value()) {
+      FSM_SEND(deal, ClientEvent::ClientEventFailed);
+      return;
+    }
+    FSM_SEND(deal, ClientEvent::ClientEventDealPublished);
   }
 
   void StorageMarketClientImpl::onClientEventDealPublished(
       std::shared_ptr<ClientDeal> deal,
       ClientEvent event,
       StorageDealStatus from,
       StorageDealStatus to) {
-    // verify deal activated - on deal sector commit
+    // TODO (a.chernyshov) verify deal activated - on deal sector commit
```

Comment:
why not implement this now?

Reply:
There is no sector builder and api method for now.